skimage/measure/_marching_cubes_lewiner.py

```diff
@@ -0,0 +1,204 @@
+from . import _marching_cubes_lewiner_cy
+
+
+
+def marching_cubes_lewiner(volume, level=None, spacing=(1., 1., 1.),
+                           gradient_direction='descent', step_size=1,
+                           allow_degenerate=True, use_classic=False):
+    """
+    Lewiner marching cubes algorithm to find surfaces in 3d volumetric data
+    
+    In contrast to ``marching_cubes()``, this algorithm resolves
+    ambiguities and guarantees topologically correct results.
+    
+    Parameters
+    ----------
+    volume : (M, N, P) array
+        Input data volume to find isosurfaces. Will internally be
+        converted to float32 if necessary.
+    level : float
+        Contour value to search for isosurfaces in `volume`. If not
+        given or None, the average of the min and max of vol is used.
+    spacing : length-3 tuple of floats
+        Voxel spacing in spatial dimensions corresponding to numpy array
+        indexing dimensions (M, N, P) as in `volume`.
+    gradient_direction : string
+        Controls if the mesh was generated from an isosurface with gradient
+        descent toward objects of interest (the default), or the opposite,
+        considering the *left-hand* rule.
+        The two options are:
+        * descent : Object was greater than exterior
+        * ascent : Exterior was greater than object
+    step_size : int
+        Step size in voxels. Default 1. Larger steps yield faster but
+        coarser results. The result will always be topologically correct
+        though.
+    allow_degenerate : bool
+        Whether to allow degenerate (i.e. zero-area) triangles in the
+        end-result. Default True. If False, degenerate triangles are
+        removed, at the cost of making the algorithm slower.
+    use_classic : bool
+        If given and True, the classic marching cubes by Lorensen (1987)
+        is used. This option is included for reference purposes. Note
+        that this algorithm has ambiguities and is not guaranteed to
+        produce a topologically correct result. The results with using
+        this option are *not* generally the same as the ``marching_cubes()``
+        function.
+    
+    Returns
+    -------
+    verts : (V, 3) array
+        Spatial coordinates for V unique mesh vertices. Coordinate order
+        matches input `volume` (M, N, P).
+    faces : (F, 3) array
+        Define triangular faces via referencing vertex indices from ``verts``.
+        This algorithm specifically outputs triangles, so each face has
+        exactly three indices.
+    normals : (V, 3) array
+        The normal direction at each vertex, as calculated from the
+        data.
+    values : (V, ) array
+        Gives a measure for the maximum value of the data in the local region
+        near each vertex. This can be used by visualization tools to apply
+        a colormap to the mesh.
+    
+    Notes about the algorithm
+    -------------------------
+    
```

Comment:
Section header should just be "Notes" to make the docscrape happy, and we can trim out this newline.